src/components/admin/form-components/withFalcorSearch.jsx

```diff
@@ -7,33 +7,30 @@ import { withFalcor } from 'components/hocs/falcor-hocs';
 import { debounce } from 'lib/utilities';
 import { getDisplayName } from 'lib/higher-order-helpers';
 
-export const withSearchableConcept = (
-  fields,
-  formatter,
-  category,
-) => WrappedField => {
+export const withFalcorSearch = (fields, formatter, mode) => WrappedField => {
   const debounceTime = 250;
   class Searchable extends React.Component {
     constructor(props) {
       super(props);
-      this.state = {
-        suggestions: [],
-      };
+      this.getSuggestions = this.getSuggestions.bind(this);
+      this.debounceSuggestions = this.debounceSuggestions.bind(this)();
+    }
 
-      this.handleChange = this.handleChange.bind(this);
-      this.handleClick = this.handleClick.bind(this);
+    getSuggestions(query) {
+      return Promise.resolve(this.debounceSuggestions(query));
+    }
 
-      this.debounceSuggestions = debounce(query => {
+    debounceSuggestions() {
```

Comment:
What I meant here was simply writing `debounceSuggestions = debounce(query => ...`. This has now become a function factory right?

Reply:
I am a bit confused here. In either case, I don't see how I can omit binding to `this`. Can you please elaborate :)?

Reply:
@emilgoldsmith

Reply:
The arrow function automatically binds this so the function you are giving to debounce is already bound, and therefore all the logic that goes on in there will have access to `this`. Feel free to ask if there's anything else confusing you. But yeah I'm pretty sure you're returning a function instead of a result in this function right?

Reply:
Got it, thanks!